go/porcelain/deploy.go

```diff
@@ -538,10 +538,17 @@ func (n *Netlify) uploadFile(ctx context.Context, d *models.Deploy, f *FileBundl
 			context.GetLogger(ctx).WithError(operationError).Errorf("Failed to upload file %v", f.Name)
 			apiErr, ok := operationError.(apierrors.Error)
 
-			if ok && apiErr.Code() == 401 {
-				sharedErr.mutex.Lock()
-				sharedErr.err = operationError
-				sharedErr.mutex.Unlock()
+			if ok {
+				if apiErr.Code() == 401 {
+					sharedErr.mutex.Lock()
+					sharedErr.err = operationError
+					sharedErr.mutex.Unlock()
+				}
+
+				// TODO this changes retry behaviour for the fileUpload case as well. OK?
+				if apiErr.Code()/100 == 4 {
+					return nil
```

Comment:
I know this is a draft PR so maybe the implementation isn't finished yet but we'll need to set the `sharedErr` too here.